Feature/table directive

ElenaBorisenkoO opened the pull request.

ElenaBorisenkoO committed:
feature/table-directive, add directive for table

davidkin commented on src/app/pages/admin/admin.controller.js:
```diff
@@ -29,8 +30,8 @@ export default class AdminController {
-        let userRole = this.rootScope.listOfUsers[id].role;
-        userRole = userRole === 'admin' ? 'user': 'admin';
+        // let userRole = this.rootScope.listOfUsers[id].role;
+        const userRole = this.rootScope.listOfUsers[id].role === 'admin' ? 'user': 'admin';
```
wat admin.controller do in feature/table directive ?

ElenaBorisenkoO replied:
because this small optimization doesn't require new branch

motifej replied:
remove this change, move to new branch and do refactor, you should set in arguments what is new role!
changeUserRole(id, newrole) {
this.syncDataService.changeUserRole(id, newrole);
}

ElenaBorisenkoO replied:
Ok. I've removed this change and will do it on a new branch.

davidkin reviewed (dismissed):
Okey, but work with admin.controller you must do fix/admin.controller

PodstDO reviewed (dismissed):
Good solution! 😍 I believe, You will rewrite `home.controller` to ES6 style in the next commit 😃

motifej commented on src/app/core/directives/table/table.directive.js:
```diff
@@ -0,0 +1,53 @@
+      restrict: 'E',
+      transclude: true,
+      template: `
+        <table class="table table-striped rounded table-hover">
```
move template to html file

ElenaBorisenkoO replied:
have done it

motifej commented on src/app/core/services/constants.js:
```diff
@@ -13,4 +13,10 @@ export default function (app) {
         'APIkey': 'b0f2a7928e6c418baf01f03c9101c5df'
     });
+    app.constant('homeConstants', {
+        'headers':  ['Amount From', 'Amount To', 'Commission', 'Currency From', 'Currency To', 'Date', 'Rate']
```
this is not a global constants, put it just into controller

ElenaBorisenkoO replied:
have done it

motifej commented on src/app/pages/admin/admin.controller.js:
```diff
@@ -12,10 +12,13 @@ export default class AdminController {
         this.listOfUsers = this.rootScope.listOfUsers;
 
         this.scope.filteredItems = [];
```
don't use scope, put all variable to `this`

ElenaBorisenkoO replied:
fixed it

ElenaBorisenkoO committed:
feature table-directive, add admin part
pass callbacks to directive

ElenaBorisenkoO committed:
feature/table-directive, add constants
remove unnecessary html from admin.html
add temperary variables for admin contriller

ElenaBorisenkoO committed:
resolve merge conflicts

pseudopilot commented on src/app/core/services/sync-data.service.js:
```diff
@@ -87,9 +87,7 @@ export default function (app) {
 
       this.changeUserRole = (uid, role) => {
-        console.log(role);
         $rootScope.listOfUsers[uid].role = role;
```
We're not able to reach the element in the `listOfUsers` by the key `uid` now, because we decided to transform listOfUsers as Object with key `uid` into Array without any keys.
Probably, we should leave original listOfUsers as an Object (as we get it from Firebase) but create a separate array for the table?

ElenaBorisenkoO replied:
Yes, it would be better to create separate array especially for admin page, because we don't need all parameters which listOfUsers contains. 
At this branch I used listOfUsers as an object because I don't have array yet. As soon as I get data in array format I'll rewrite particular functions.

ElenaBorisenkoO committed:
feature/table-directive, replace this.scope with this

ElenaBorisenkoO committed:
feature/table-directive, return constants to headcontrollers
and admin controller

ElenaBorisenkoO committed:
feature/table-directive, move table template to
separate html file

ElenaBorisenkoO committed:
feature/table-directive, return old variant to admin controller

ElenaBorisenkoO committed:
feature/table-directive, fix merge conflicts
